adobecom/milo
# MWPW-150665: [POC] Spectrum 2 Styles and Spectrum CSS Tokens Integration Script

This PR serves as a POC for ingesting Spectrum 2 Tokens into Milo. It introduces a script designed to extract and format Spectrum CSS Token values and integrate them into a single Spectrum 2 Milo CSS file.

* **Introduces Spectrum 2 Milo CSS styles:** This file includes all the core required Spectrum 2 custom properties for Milo. It consists of Milos specfic S2 custom properties sourced from [Spectrum CSS Tokens*](https://github.com/adobe/spectrum-css/tree/main/tokens) and "Narrative" values that only exist in Milo.
    * `--spectrum` prefixed custom properties are properties sourced from Spectrum CSS Tokens
    * `--sn` prefixed custom properties are "Narrative" properties specific to Milo

* **Automated Spectrum 2 Token Script:** A script is added to streamline the process of updating Spectrum 2 Milo CSS styles with S2 Token values. This script extracts and formats S2 values from Spectrum CSS Tokens, and updates the corresponding values in the Spectrum 2 Milo CSS file.

**Demonstrated Concept:**

This POC demonstrates a viable option for integrating Spectrum Tokens into Milo. It formats and extracts only the essential custom properties from the original 25kB gzipped Spectrum Tokens file resulting in a leaner 5.5kB Milo CSS file gzipped that's more in alignment with Milo's performance standards.

Resolves: [MWPW-150665](https://jira.corp.adobe.com/browse/MWPW-15066)

**Validation**
- Verify custom properties in `s2-styles.css` align with [Milo Spectrum 2 Values from design](https://docs.google.com/spreadsheets/d/18SdT8J2xkP4gVqWLJFXE2koPwRCHMNrUJ5NNsRmfH9k/edit#gid=2131325156)
- Verify working build by changing the values of `--s2` prefixed custom properties. 
    - Run `npm run build:spectrum2-styles` 
    - Verify the build automatically updates the custom property to the correct value as specified in the document in the previous step.




*Spectrum 2 CSS Tokens are currently sourced from Spectrum CSS tokens `v14.0.0-next.7` beta branch.

URL for testing:
- https://saasha-mwpw-150665-spectrum2-poc--milo--adobecom.hlx.page/
